website/src/theme/SearchBar/index.js

```diff
@@ -1,207 +1,114 @@
-
-let DocSearchModal = null;
-
-function Hit({hit, children}) {
-  return <Link to={hit.url}>{children}</Link>;
-}
-
-function ResultsFooter({state, onClose}) {
-  const {generateSearchPageLink} = useSearchQuery();
-
-  return (
-    <Link to={generateSearchPageLink(state.query)} onClick={onClose}>
-      See all {state.context.nbHits} results
-    </Link>
-  );
-}
-
-function DocSearch({contextualSearch, ...props}) {
-  const {siteMetadata} = useDocusaurusContext();
-
-  const contextualSearchFacetFilters = useAlgoliaContextualFacetFilters();
-
-  const configFacetFilters = props.searchParameters?.facetFilters ?? [];
-
-  const facetFilters = contextualSearch
-    ? // Merge contextual search filters with config filters
-      [...contextualSearchFacetFilters, ...configFacetFilters]
-    : // ... or use config facetFilters
-      configFacetFilters;
-
-  // we let user override default searchParameters if he wants to
-  const searchParameters = {
-    ...props.searchParameters,
-    facetFilters,
-  };
-
-  const {withBaseUrl} = useBaseUrlUtils();
+import React, { useRef, useCallback, useState } from "react";
+import classnames from "classnames";
+import { useHistory } from "@docusaurus/router";
+import useDocusaurusContext from "@docusaurus/useDocusaurusContext";
+import { usePluginData } from '@docusaurus/useGlobalData';
+import useIsBrowser from "@docusaurus/useIsBrowser";
+const Search = props => {
+  const initialized = useRef(false);
+  const searchBarRef = useRef(null);
+  const [indexReady, setIndexReady] = useState(false);
   const history = useHistory();
-  const searchContainer = useRef(null);
-  const searchButtonRef = useRef(null);
-  const [isOpen, setIsOpen] = useState(false);
-  const [initialQuery, setInitialQuery] = useState(null);
-
-  const importDocSearchModalIfNeeded = useCallback(() => {
-    if (DocSearchModal) {
-      return Promise.resolve();
-    }
-
-    return Promise.all([
-      import('@docsearch/react/modal'),
-      import('@docsearch/react/style'),
-      import('./styles.css'),
-    ]).then(([{DocSearchModal: Modal}]) => {
-      DocSearchModal = Modal;
-    });
-  }, []);
-
-  const onOpen = useCallback(() => {
-    importDocSearchModalIfNeeded().then(() => {
-      searchContainer.current = document.createElement('div');
-      document.body.insertBefore(
-        searchContainer.current,
-        document.body.firstChild,
-      );
-      setIsOpen(true);
-    });
-  }, [importDocSearchModalIfNeeded, setIsOpen]);
-
-  const onClose = useCallback(() => {
-    setIsOpen(false);
-    searchContainer.current.remove();
-  }, [setIsOpen]);
-
-  const onInput = useCallback(
-    (event) => {
-      importDocSearchModalIfNeeded().then(() => {
-        setIsOpen(true);
-        setInitialQuery(event.key);
+  const { siteConfig = {} } = useDocusaurusContext();
+  const isBrowser = useIsBrowser();
+  const { baseUrl } = siteConfig;
+  const initAlgolia = (searchDocs, searchIndex, DocSearch) => {
```

Comment:
seems like several references to algolia but seems to be completely replaced by lunr

Reply:
> it looks like the creator of the lunr-search plugin forked the algolia search implementation (see the note on their readme [here](https://github.com/praveenn77/docusaurus-lunr-search#disableversioning)). So it might be possible to fully remove references to algolia from the swizzled component, but IMO that seems like a low-pri optimization (would require testing, could introduce maintenance friction as we'd be deviating from the canonical package, etc).